src/app/home-page/recent-events/recent-events.component.ts

```diff
@@ -57,15 +98,48 @@ export class RecentEventsComponent implements OnInit {
           this.alertService.detailedError(error);
         }
       );
+      this.getAUsersEvents();
+      this.loading$ = this.recentEventsQuery.selectLoading();
     } else {
-      this.recentEventsService.get();
+      // On dashboard, get my events
+      this.isLoading = true;
+      this.supportedEventTypes = this.eventType === 'SELF_ORGANIZATIONS' ? this.supportedOrgEventTypes : this.supportedStarredEventTypes;
+      this.getMyEventsByType();
     }
+  }
+
+  /**
+   * Get another user's events when viewing their user page
+   */
+  private getAUsersEvents() {
+    this.recentEventsQuery.selectAll({ filterBy: (entity) => this.supportedUserEventTypes.includes(entity.type) }).subscribe(
+      (events) => {
+        this.events = events;
+      },
+      (error: HttpErrorResponse) => {
+        this.alertService.detailedError(error);
+      }
+    );
+  }
 
-    this.events$ = this.recentEventsQuery.selectAll({
-      filterBy: (entity) => this.supportedEventTypes.includes(entity.type),
-    });
-    this.loading$ = this.recentEventsQuery.selectLoading();
-    this.noEvents$ = this.events$.pipe(map((events) => !events || events.length === 0));
+  /**
+   * Get my events based on eventType input
+   */
+  private getMyEventsByType() {
+    this.eventsService
+      .getEvents(this.eventType)
+      .pipe(
+        finalize(() => (this.isLoading = false)),
+        takeUntil(this.ngUnsubscribe)
+      )
+      .subscribe(
+        (events) => {
+          this.events = events.filter((event) => this.supportedEventTypes.includes(event.type)).slice(0, 4);
```

Comment:
Do you intentionally slice here and intentionally **not** slice on line 117? If no, then create a method to handle both cases to work the same and avoid the duplication.

Another option is do the slicing in the HTML with the Slice pipe. I guess it depends how many events you're expecting; if you expecting hundreds of thousands, then yeah probably best to not keep them all around (and ideally we would handle that in the API so you don't get hundreds of thousands into the browser in the first place). 

I could go either way, and feel free to keep it this way, but I'd lean towards the data being the data, and choosing what portion of it to display being a UI option. Unless, like I said, the data gets to be an unmanageable size.

Reply:
Yeah its intentional, I limit to a max of 4 events for the dashboard boxes, but for the user pages we can extend to fit all of the events (returns a max of 10).  It looks like both `selectAll()` and `getEvents()` return a max of 10 events so there's no risk in receiving hundreds.

I'm adding a `displayLimit` member to slice in the HTML and creating a `filterByEventType` method to handle filtering for both cases 👍️